File: backend/src/main/java/com/pjatk/QuizzApp/Quiz/Quiz.java
package com.pjatk.QuizzApp.Quiz;

import com.pjatk.QuizzApp.Question.Question;
import com.pjatk.QuizzApp.User.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.Duration;
import java.time.Instant;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Quiz
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String title;
    private String description;
    @Enumerated(EnumType.STRING)
    private Category category;
    @Enumerated(EnumType.STRING)
    private Diff diff;
    @Lob
    private byte[] image;
    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false)
    private User author;
    private Instant createdAt = Instant.now();
    private Integer timeLimit;
    private Integer maxPoints;

    @OneToMany(mappedBy = "quiz", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Question> questions = new ArrayList<>();



}

File: frontend/src/types/quiz.ts
import type {Diff} from "@/types/diff.ts";
import type {User} from "@/types/user.ts";
import type {Question} from "@/types/question.ts";

export interface Quiz {
  id: number | null; // null for new unsaved quizzes
  title: string;
  description: string;
  category: string;
  diff: Diff;
  image: string;
  author: User | number; // Use ID or minimal user info
  createdAt: string; // ISO string (Instant)
  timeLimit: string; // ISO string (LocalDateTime)
  finishedByUsers: User[] | number[]; // Use IDs
  maxPoints: number;
  userScore: User[] | number[]; // Use IDs
  questions?: Question[]; // Optional for nested data
}
